src/aws-cpp-sdk-core/include/smithy/tracing/TraceMessageEvent.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#pragma once
+
+#include <smithy/Smithy_EXPORTS.h>
+#include <smithy/tracing/TraceEventDataInterface.h>
+#include <aws/core/utils/memory/stl/AWSString.h>
+
+#include <utility>
+
+namespace smithy {
+    namespace components {
+        namespace tracing {
+            /**
+             * A Trace event that is a message event.
+             */
+            class SMITHY_API TraceMessageEvent : public TraceEventDataInterface {
+            private:
+                Aws::String message;
+            public:
+                explicit TraceMessageEvent(Aws::String message) :
+                    message(std::move(message)) {}
```

Comment:
nit: string is not passed by ref, what is move trying to do?

Reply:
[clang tidy recommended pass by move](https://clang.llvm.org/extra/clang-tidy/checks/modernize/pass-by-value.html)
 copying from their reasoning

With move semantics added to the language and the standard library updated with move constructors added for many types it is now interesting to take an argument directly by value, instead of by const-reference, and then copy. This check allows the compiler to take care of choosing the best way to construct the copy.

The transformation is usually beneficial when the calling code passes an rvalue and assumes the move construction is a cheap operation.`